src/pgwire/src/server.rs

```diff
@@ -138,7 +140,24 @@ impl Server {
                                 conn_uuid_handle.set(conn_uuid);
                                 debug!(conn_uuid = %conn_uuid_handle.display(), conn_uuid_forwarded, "starting new pgwire connection in adapter");
 
-                                let mut conn = FramedConn::new(conn_id.clone(), conn);
+                                let direct_peer_addr = conn.inner_mut().peer_addr().map(|addr| addr.ip()).expect("pgwire connection must have peer address");
```

Comment:
Same comment as in the HTTP server—better and hopefully easy to gracefully return an error if the `peer_addr()` is somehow missing.